src/devices/ExplorerHat/Lights.cs

```diff
@@ -115,10 +119,13 @@ protected virtual void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    LedArray[0].Dispose();
-                    LedArray[1].Dispose();
-                    LedArray[2].Dispose();
-                    LedArray[3].Dispose();
+                    if (_shouldDispose)
```

Comment:
NIT, looks like we forgot to also dispose the controller here.

Reply:
the controller should not be disposed in this dispose as every individual Led is disposing it. I moved out of the if the individual calls to each Led.

Reply:
would that be a problem though? it will mean that perhaps the first LED will dispose it, and then the second one will try to call Off() before disposing it again and that will fail. Wouldn't it be better to just have the LEDs dispose and have the Ligths Dispose dispose the actual controller?

Reply:
Yes, great catch. Adding Off(); before disposing. Then the controller will be disposed at the Lights level, not at every individual level.